src/components/bannerCard.tsx

```diff
@@ -0,0 +1,246 @@
+                                }}
+                            />
+                        </FormItem>
+                        <ExpiryPicker
+                            expiryDate={this.state.expirationDate}
+                            onChange={(expirationDate) => {
+                                this.setState({ expirationDate, expiryErrorText: null });
+                                this.ensureMarkedDirty();
+                            }}
+                            onParseError={(errorMessage) => {
+                                this.setState({ expiryErrorText: errorMessage });
+                            }}
+                        />
+                    </div> : null}
+                </div>
+            </Card>
+        );
+    }
+
+    private getMessageError(message: string): string {
+        if (message.split(" ").length >= 30) {
+            return "Message too long";
+        }
+
+        return null;
+    }
+
+    private async saveBanner(): Promise<void> {
+        this.setState({ loading: true });
+
+        const banner = this.props.banner;
+        banner.message = this.state.message;
+        banner.level = this.state.level;
+        banner.expirationDate = this.state.expirationDate;
+
+        try {
+            await VstsService.instance.saveWebGlobalMessageBanner(banner);
+
+            this.props.onSave(banner);
+
+            this.setState({ dirty: false, loading: false });
+
+            this.isSaveEnabled();
+        } catch (ex) {
+            this.setState({ messageErrorText: "Unable to save" });
+        }
+
+    }
+
+    private async deleteBanner(): Promise<void> {
+        try {
+            await VstsService.instance.deleteWebGlobalMessageBanner(this.props.banner);
+
+            this.props.onDelete();
+        } catch (ex) {
+            this.setState({ messageErrorText: "Unable to delete" });
+        }
+    }
+
+    private isSaveEnabled(): boolean {
+        return this.state.dirty
+            && this.errorText == null
+            && !this.isMessageExpired();
+    }
+
+    private ensureMarkedDirty(): void {
+        if (this.state.dirty === false) {
+            this.setState({ dirty: true });
+        }
+    }
+
+    private isMessageExpired(): boolean {
+        return this.state.expirationDate != null && (this.state.expirationDate < new Date());
+    }
+
+    private setupFields(): void {
+        if (this.props.banner == null) {
+            return;
+        }
+
+        this.setState(this.props.banner);
+    }
+
+    private getStatusProps(): IStatusProps {
+        if (this.errorText != null) {
+            return {
+                ...Statuses.Failed,
+                text: this.errorText,
+            };
+        }
+
+        if (!this.state.dirty) {
+            if (this.state.expirationDate == null) {
+                return {
+                    ...Statuses.Success,
+                    text: "This message will be shown indefinitely.",

```

Comment:
Nit: "This" or "The" ?

Reply:
It's a general nit for this method (there are multiple strings ...) :)

Reply:
Agreed. "The"